common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -364,18 +364,18 @@ public void record(Object hint) {
         }
 
         private void record0(Object hint, int recordsToSkip) {
-            if (creationRecord != null) {
+            if (creationRecord != null && MAX_RECORDS > 0) {
```

Comment:
1. If `MAX_RECORDS == 0` and `lastRecords.size() == 0`, then `size >= MAX_RECORDS` is true, `lastRecords.removeFirst()` will raise error. Even if we check `size > 0 && size >= MAX_RECORDS` before remove, there will always be one record added to lastRecords.

2. I don't get this "Before the user could use MAX_RECORDS = 0 to basically enforce there is no limit at all.". I can't find any special check for `MAX_RECORDS = 0`.